src/ert/job_runner/reporting/event.py

```diff
@@ -53,15 +53,20 @@ def __init__(self, evaluator_url, token=None, cert_path=None):
         self._step_id = None
         self._event_queue = queue.Queue()
         self._event_publisher_thread = threading.Thread(target=self._publish_event)
+        self._sentinel = object()  # notifying the queue's ended
 
     def _publish_event(self):
         logger.debug("Publishing event.")
         with Client(self._evaluator_url, self._token, self._cert) as client:
             while True:
                 event = self._event_queue.get()
-                if event is None:
+                if event is self._sentinel:
+                    self._event_queue.task_done()
                     return
-                client.send(to_json(event).decode())
+                try:
+                    client.send(to_json(event).decode())
+                finally:
+                    self._event_queue.task_done()
```

Comment:
We haven't completely decided on the behaviour of sending events... But some comments:

Here the exception will be thrown after the `finally`, hence for any remaining events in the queue, they will not be completed with `task_done`. This was an issue before you had the `finally` anyway, so nice that we look over it. Could be tested by running the `job_dispatch` without setting up any server. (should be practical to make as a test as well, just make sure that we don't have to wait for the entire backoff time before it fails. Possible to set number of retries and time length on the `Client`).

Please do a check on the expectation from ert if a job never sends its finished event, will the jobqueuemanager fail on `MAX_RUNTIME` for a single job?

Reply:
We've abandoned marking queue task_done since we might never finish up the queue processing. Current implementation just looses events if no connection's been made and such automatically closes, when the `Finish` event was received.